MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -2134,35 +2137,36 @@ private string GetFixedSizeScalingFilter(Video3DFormat? threedFormat, int reques
             {
                 if (isVaapiH264Encoder)
                 {
-                    filters.Add(string.Format(CultureInfo.InvariantCulture, "deinterlace_vaapi"));
+                    filters.Add(
+                        string.Format(
+                            CultureInfo.InvariantCulture,
+                            "deinterlace_vaapi=rate={0}",
+                            doubleRateDeinterlace ? "field" : "frame"));
                 }
             }
 
             // Add software deinterlace filter before scaling filter
-            if (state.DeInterlace("h264", true)
+            if ((state.DeInterlace("h264", true)
                 || state.DeInterlace("avc", true)
                 || state.DeInterlace("h265", true)
                 || state.DeInterlace("hevc", true))
+                && (!isVaapiH264Encoder && !isQsvH264Encoder && !isNvdecH264Decoder))
```

Comment:
Not sure which is better. Maybe @Bond-009 has an opinion.
```suggestion
                && !isVaapiH264Encoder
                && !isQsvH264Encoder
                && !isNvdecH264Decoder)
```

Reply:
I've made this change.